src/plugins/.common/structures/DataHandling/DatabaseHandler.js

```diff
@@ -1,13 +1,246 @@
+    invites.splice(invites.indexOf(invite), 1);
+    this._setInvites(guildId, invites);
+    return true;
+  }
+
+  async setInvite(guildId, name, code, createIfNotExist = true) {
+    const invites = await this.getInvites(guildId, createIfNotExist);
+    const index = invites.findIndexOf((invite) => invite.name == name);
+    if (index === -1) return false;
+
+    const invite = invites[index];
+    //If the invite doesn't exist or no fields are going to be changed, return false;
+    if (!invite || (invite.name === name && invite.code === code)) return false;
+
+    invites[index].name = name;
+    invites[index].code = code;
+    this._setInvites(invites);
+    return true;
+  }
+
+  async _setInvites(guildId, invites) {
+    const guild = await this._guild.getGuild(guildId);
+    guild.invites = invites;
+    guild.save();
+  }
+}
+
+class ChannelsHandler {
+  constructor() {
+    this._guild = new GuildHandler();
+  }
+
+  createChannels() {}
+
+  async getChannels(guildId, createIfNotExist = true) {
+    const customGuildId = await this._guild.getId(guildId, createIfNotExist);
+    if (!customGuildId) return null;
+    const channels = await Channel.findOne({ where: { guildId: customGuildId } });
+    if (channels) return channels;
+    return await Channel.create({ guildId: customGuildId });
+  }
+
+  //DO NOT USE THIS METHOD UNLESS NECESSARY. Use the specific field methods below!
+  async getField(guildId, field, createIfNotExist = true) {
+    const channels = await this.getChannels(guildId, createIfNotExist);
+    if (!channels) return;
+    return channels.get(field);
+  }
+
+  async getWelcome(guildId, createIfNotExist = true) {
+    return await this.getField(guildId, "welcome", createIfNotExist);
+  }
+
+  async getOpportunity(guildId, createIfNotExist = true) {
+    return await this.getField(guildId, "opportunity", createIfNotExist);
+  }
+
+  async getAuditLog(guildId, createIfNotExist = true) {
+    return await this.getField(guildId, "auditLog", createIfNotExist);
+  }
+
+  async getMemberCounter(guildId, createIfNotExist = true) {
+    return await this.getField(guildId, "memberCounter", createIfNotExist);
+  }
+  async getThreadDirectory(guildId, createIfNotExist = true) {
+    return await this.getField(guildId, "threadDirectory", createIfNotExist);
+  }
+
+  async getTeamsAndProjects(guildId, createIfNotExist = true) {
+    return await this.getField(guildId, "teamsAndProjects", createIfNotExist);
+  }
+
+  async getNewUserMention(guildId, createIfNotExist = true) {
+    return await this.getField(guildId, "newUserMention", createIfNotExist);
+  }
+
+  async getIntroductions(guildId, createIfNotExist = true) {
+    return await this.getField(guildId, "introductions", createIfNotExist);
+  }
+
+  async getInvites(guildId, createIfNotExist = true) {
+    return await this.getField(guildId, "invites", createIfNotExist);
+  }
+
+  async getMinecraft(guildId, createIfNotExist = true) {
```

Comment:
I feel like the getter and setter methods are making the code really long. Is it really necessary to create a get and set method for each column?